src/app/static/tests/unit/components/parameters/parameters.test.ts

```diff
@@ -120,6 +138,92 @@ describe("Parameters", () => {
+        expect(spans.at(1).text()).toBe("2.00m");
+    });
+
+    it("renders population in thousand", async () => {
+        const wrapper = getWrapper(paramValuesIreland);
+
+        const populationProp = numericFormatter(21000);
+        await wrapper.setProps({ population: populationProp });
+
+        const countryDiv = wrapper.find("#countries");
+        expect(countryDiv.find("label").text()).toBe("Country");
+
+        const countrySelect = countryDiv.find("v-select-stub");
+        expect(countrySelect.props("value").code).toBe("IRE");
+
+        const populationDiv = wrapper.find("#population");
+        const spans = populationDiv.findAll("span");
+        expect(spans.at(0).text()).toBe("Population:");
+        expect(spans.at(1).text()).toBe("21.00k");
+    });
+
+    it("renders population in hundred", async () => {
+        const wrapper = getWrapper(paramValuesIreland);
+        const populationProp = numericFormatter(100);
+        await wrapper.setProps({ population: populationProp });
+
+        const countryDiv = wrapper.find("#countries");
+        expect(countryDiv.find("label").text()).toBe("Country");
+
+        const countrySelect = countryDiv.find("v-select-stub");
+        expect(countrySelect.props("value").code).toBe("IRE");
+
+        const populationDiv = wrapper.find("#population");
+        const spans = populationDiv.findAll("span");
+        expect(spans.at(0).text()).toBe("Population:");
+        expect(spans.at(1).text()).toBe("100");
+    });
+
+    it("renders population in zero", async () => {
+        const wrapper = getWrapper(paramValuesIreland);
+        const populationProp = numericFormatter(-1);
+        await wrapper.setProps({ population: populationProp });
+
+        const countryDiv = wrapper.find("#countries");
+        expect(countryDiv.find("label").text()).toBe("Country");
+
+        const countrySelect = countryDiv.find("v-select-stub");
+        expect(countrySelect.props("value").code).toBe("IRE");
+
+        const populationDiv = wrapper.find("#population");
+        const spans = populationDiv.findAll("span");
+        expect(spans.at(0).text()).toBe("Population:");
+        expect(spans.at(1).text()).toBe("0");
+    });
+
+    it("renders population in billion", async () => {
+        const wrapper = getWrapper(paramValuesIreland);
+        const populationProp = numericFormatter(5000000000);
+        await wrapper.setProps({ population: populationProp });
+
+        const countryDiv = wrapper.find("#countries");
+        expect(countryDiv.find("label").text()).toBe("Country");
+
+        const countrySelect = countryDiv.find("v-select-stub");
+        expect(countrySelect.props("value").code).toBe("IRE");
+
+        const populationDiv = wrapper.find("#population");
+        const spans = populationDiv.findAll("span");
+        expect(spans.at(0).text()).toBe("Population:");
+        expect(spans.at(1).text()).toBe("5.00b");
+    });
+
```

Comment:
This feels like the wrong place to be testing the numeric formatting works for all the different types of numbers. For the parameters component, the population value is just a string which it renders without manipulating in any way so I think you just need one test for that, and test all the numeric permutations in direct unit tests of `numericFormatter` - those should be much simpler tests to write.